### PR TITLE
[docs] Describe merge_query

### PR DESCRIPTION
**Pull request by aquamatthias**

### CODE REVIEW

File: docs/source/manual/discovery.rst [hunk 1]
```diff
@@ -57,66 +63,41 @@ We advise to use ``help`` in the :ref:`component-cksh` CLI to get more informati
...
-
-  You can also use the command ``match`` as an abbreviation for ``reported``:
-
-  ``match kind == aws_account``
+  ``query is(account)``
 
-desired
--------
+  This will select all accounts of all collected cloud providers.
+  In oder to select only accounts from AWS, we could achieve this with:
```
Comment (TheCatLady) [inline docs/source/manual/discovery.rst:74]:
```suggestion
  In order to select only accounts from AWS, we could achieve this with:
```

File: docs/source/manual/discovery.rst [hunk 2]
```diff
@@ -57,66 +63,41 @@ We advise to use ``help`` in the :ref:`component-cksh` CLI to get more informati
...
-  You can also set properties in the ``desired`` section by using the ``set_desired`` command:
-
-  ``set_desired clean = false``
-
-  An example, selecting all graph nodes that are marked for cleanup and reversing this by chaining the two commands using a pipe:
+  ``query is(aws_account)``
 
-  ``desired clean == true  | set_desired clean = false``
+  It is possible to filter all found accounts by available properties.
+  Let us filter aws accounts, that are either older than 2 weeks.
```
Comment (TheCatLady) [inline docs/source/manual/discovery.rst:79]:
```suggestion
  Let us filter AWS accounts, that are either older than 2 weeks.
```

**Review by TheCatLady — COMMENTED** [review]
Just skimmed over the changes and noticed a couple of mistakes. I'll proofread and edit as I migrate the docs to the new repo, though!

**Comment by aquamatthias** [comment]
Thanks @TheCatLady .

File: docs/source/manual/discovery.rst [hunk 3]
```diff
@@ -26,18 +26,24 @@ A graph node is a json document with a well defined structure and these top leve
...
       "id": "xxx",
-      "kinds": [ ... ],
       "reported": { ... },
       "desired": { ... },
       "metadata": { ... }
     }
 
 
-Each graph node always has an ``id`` that is a unique id created by Cloudkeeper as well as a ``kinds`` array containing the kind of the node including all it's parents kinds.
+Each graph node always has an ``id`` that is a unique id created by Cloudkeeper.
```
Comment (TheCatLady) [inline docs/source/manual/discovery.rst:35]:
```suggestion
Each graph node always has an ``id`` that is a unique ID created by Cloudkeeper.
```

File: docs/source/manual/discovery.rst [hunk 4]
```diff
@@ -26,18 +26,24 @@ A graph node is a json document with a well defined structure and these top leve
...
 
-It also features three main sections: ``reported``, ``desired`` and ``metadata``.
+The ``reported`` section contains data reported from the specific cloud provider.
+The content and schema of the ``reported`` section is defined by the cloud provider and described formally in the :ref:`model`.
 
-You learn more about the structure and sections in our :ref:`graph_node_spotlight`.
+The ``desired`` section can be manipulated by users and tools to mark and trigger an intended change on the
+specific resource. Cleaning up a resource for example is done by setting ``clean=true`` in the desired section.
```
Comment (TheCatLady) [inline docs/source/manual/discovery.rst:41]:
```suggestion
specific resource. Cleaning up a resource, for example, is done by setting ``clean=true`` in the desired section.
```

File: docs/source/manual/discovery.rst [hunk 5]
```diff
@@ -26,18 +26,24 @@ A graph node is a json document with a well defined structure and these top leve
...
+
+The ``metadata`` section contains additional data of a resource, that is not reported by the cloud provider,
+but added by the Cloudkeeper tool chain.
```
Comment (TheCatLady) [inline docs/source/manual/discovery.rst:44]:
```suggestion
The ``metadata`` section contains additional data about the resource. This data is not from the cloud provider,
but added by the Cloudkeeper toolchain.
```

File: docs/source/manual/discovery.rst [hunk 6]
```diff
@@ -26,18 +26,24 @@ A graph node is a json document with a well defined structure and these top leve
...
+
+See an example and learn more about the structure in :ref:`graph_node_spotlight`.
```
Comment (TheCatLady) [inline docs/source/manual/discovery.rst:46]:
```suggestion
You can see an example and learn more about the structure in :ref:`graph_node_spotlight`.
```

File: docs/source/manual/discovery.rst [hunk 7]
```diff
@@ -57,66 +63,41 @@ We advise to use ``help`` in the :ref:`component-cksh` CLI to get more informati
...
-  You can also set properties in the ``desired`` section by using the ``set_desired`` command:
-
-  ``set_desired clean = false``
-
-  An example, selecting all graph nodes that are marked for cleanup and reversing this by chaining the two commands using a pipe:
+  ``query is(aws_account)``
 
-  ``desired clean == true  | set_desired clean = false``
+  It is possible to filter all found accounts by available properties.
+  Let us filter AWS accounts, that are either older than 2 weeks.
```
Comment (TheCatLady) [inline docs/source/manual/discovery.rst:79]:
```suggestion
  It is also possible to filter results by their available properties.
  For example, let us find all AWS accounts that are more than 2 weeks old:
```

File: docs/source/manual/discovery.rst [hunk 8]
```diff
@@ -57,66 +63,41 @@ We advise to use ``help`` in the :ref:`component-cksh` CLI to get more informati
...
+  As you can see multiple criteria can be combined with ``and`` but also with ``or``.
+  In order to define precedence, it is also possible to use parentheses.
+  To filter aws accounts, that are either older than 2 weeks or have more than 10 users, we would do the following:
 
-.. admonition:: Example
+  ``query is(aws_account) and (reported.age>2w or reported.users>1)``
 
-  ``metadata ancestors.cloud.name == aws``
+  All resources that match the defined criteria will be returned.
+  It is possible to limit the result via ``limit``.
```
Comment (TheCatLady) [inline docs/source/manual/discovery.rst:90]:
```suggestion
  By default, all resources that satisfy the defined criteria will be returned.
  However, it is possible to limit the number of results by specifying ``limit``:
```

File: docs/source/manual/discovery.rst [hunk 9]
```diff
@@ -57,66 +63,41 @@ We advise to use ``help`` in the :ref:`component-cksh` CLI to get more informati
...
 
-  This will select all graph nodes that are part of AWS clouds
+  ``query is(aws_account) and (reported.age>2w or reported.users>1) limit 3``
 
-  You can set metadata directly or add custom metadata by using the ``set_metadata`` command.
+  In addition to the limit, it is also possible to define a sort criteria.
+  Following query will filter all aws accounts that are older than 2 weeks,
+  and return the 3 accounts with most users.
```
Comment (TheCatLady) [inline docs/source/manual/discovery.rst:96]:
```suggestion
  In addition to ``limit``, it is also possible to define the sort order.
  The following query will return the 3 AWS accounts that are more than 2 weeks old
  with the most of users.
```

File: docs/source/manual/discovery.rst [hunk 10]
```diff
@@ -186,15 +155,40 @@ The ``operation`` is one of the following options:
 | ``in`` : Property is one of the following values. The value has to be an array.
 | ``not in`` : Property is not one of the following values. The value has to be an array.
 
+
 value
 ^^^^^
 
 The ``value`` can be *any* json literal or *any* json conform value.
+A json conform value is:
```
Comment (TheCatLady) [inline docs/source/manual/discovery.rst:163]:
```suggestion
A JSON conform value is:
```

**Review by TheCatLady — COMMENTED** [review]
Some more suggestions :)

**Pull request merged**